cmd/oceantv/broadcast_hardware_machine.go

```diff
@@ -238,22 +242,39 @@ func (sm *hardwareStateMachine) handleTimeEvent(t timeEvent) {
 	eventIfStatus := func(e event, status bool) {
 		sm.ctx.camera.publishEventIfStatus(e, status, sm.ctx.cfg.CameraMac, sm.ctx.store, sm.log, sm.ctx.bus.publish)
 	}
-	switch sm.currentState.(type) {
+	switch state := sm.currentState.(type) {
 	case *hardwareStarting:
-		withTimeout := sm.currentState.(stateWithTimeout)
-		if withTimeout.timedOut(t.Time) {
+		if state.timedOut(t.Time) {
 			sm.ctx.bus.publish(hardwareStartFailedEvent{})
 			sm.transition(newHardwareOff())
 			return
 		}
 		eventIfStatus(hardwareStartedEvent{}, true)
 	case *hardwareStopping:
+		if state.timedOut(t.Time) {
+			if state.PreviouslyTimedOut {
+				sm.log("hardwareStopping state timed out for the second time")
+				sm.ctx.bus.publish(hardwareStopFailedEvent{})
+				sm.transition(newHardwareOn())
```

Comment:
Why are we transitioning back to HardwareOn?

Reply:
Good point. You're suggesting we stay in the stopping state, even if it fails. I think I agree. I'll have a think about this.